webapp/src/test/java/com/plhal/ares/webapp/EmbeddedPostgresTest.java

```diff
@@ -0,0 +1,41 @@
+package com.plhal.ares.webapp;
+
+import com.plhal.ares.parser.Firma;
+import com.plhal.ares.service.DataService;
+import io.zonky.test.db.AutoConfigureEmbeddedDatabase;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Class for application testing.
+ */
+@DataJpaTest
+@AutoConfigureEmbeddedDatabase
+public class EmbeddedPostgresTest {
+
+    private final DataService dataService;
```

Comment:
Unitni test by se mel zamerovat na overeni funkncnosti unity a dalsi zavislosti by mel odstinit (tedy "mockovat"). Soucasne plati, ze unitni test nema byt zavisly na externich zdrojich. To tedy napriklad znamena, ze kdyz odpojite svuj pocitac od site, tak by test mel stale dobehnout a overit funkcnost testovane unity. Protoze bez pristupu k internetu byste ani nebyl schopny projekt sestavit a otestovat. Tento test overuje funkcnost servnisni tridy (neresi jen db, ale i parsovani), coz bych nechal na samostatny ukol (bude treba vice poresit testovaci kontext a "mockovat" patricne tridy). Bohate bude stacit, kdyz v ramci tohoto ukolu otestujete databazove rozhrani. V nasem pripade tedy `FirmaRepository` a napriklad ulozeni (save), nacteni firmy.

Reply:
Soucasne nezapomente test umistit do spravneho modulu. I v tento okamzik je umisten spatne. Testuje servisni tridu a neni v servisnim modulu. To znamena, ze databazovy test ma byt v databazovem modulu.